src/microvm/src/vmm/microvm/mod.rs

```diff
@@ -371,9 +401,75 @@ impl Vmm {
+            Ok(command) => match command {
+                ControlCommand::StartMicroVm => {
+                    if self.orchestrator_state == OrchestratorState::PreBoot {
+                        // TODO: separate starting logic from `spawn()` and put it here
+                        // This TODO could be done right now, but it's a major refactor.
+                        self.orchestrator_state = OrchestratorState::Running;
+                        trace!("OrchestratorState: PreBoot -> Running");
+                    }
+                    Ok(())
+                },
+                ControlCommand::LoadSnapshotAndRun => {
+                    if self.orchestrator_state == OrchestratorState::PreBoot {
+                        // TODO: load snapshot
+                        // This TODO requires being able to create snapshots.
+
+                        // The Linux daemon should send messages to PreBoot VMMs by default,
+                        // so there's no need to tell it to resume sending messages.
+
+                        self.resume_microvm()?;
+                        trace!("OrchestratorState: PreBoot -> Running");
+                    }
+                    Ok(())
+                },
+                ControlCommand::PauseMicroVm => {
+                    if self.orchestrator_state == OrchestratorState::Running {
+                        self.pause_protocol()?;
+                    }
+                    Ok(())
+                },
+                ControlCommand::PauseAndCreateSnapshot => {
+                    if self.orchestrator_state == OrchestratorState::Running {
+                        self.pause_protocol()?;
+                        self.create_snapshot()?;
+                    }
+                    Ok(())
+                },
+                ControlCommand::CreateSnapshot => {
+                    if self.orchestrator_state == OrchestratorState::Paused {
+                        self.create_snapshot()?;
+                    }
+                    Ok(())
+                },
+                ControlCommand::ResumeMicroVm => {
+                    if self.orchestrator_state == OrchestratorState::Paused {
+                        // TODO: tell linuxd it's fine to send more messages
+                        // This TODO requires having a control plane connection with linuxd
+                        self.resume_microvm()?;
+                        trace!("OrchestratorState: Paused -> Running");
+                    }
+                    Ok(())
+                },
+                ControlCommand::LinuxDaemonFlushed => {
+                    // NOTE: this will be unreachable once the communication is fully implemented
+                    // `LinuxDaemonFlushed` should only be sent in the middle of `pause_protocol`.
+                    // In fact, it should already be unreachable, but it cannot be tested ATM.
+                    Ok(())
+                },
+            },
```

Comment:
If either `pause_protocol()`, `create_snapshot()` or `resume_microvm()`, how is gracefully termination handled?